### PR TITLE
Fix StartsWithPortableAndBootstraperExe

### PR DESCRIPTION
**Pull request by jkotalik**
For https://github.com/aspnet/AspNetCore-Internal/issues/1895

The cause for this failure was due to publish changes we made. The published runtimeconfig.json file we created for standalone applications was being copied to the published output for portable exe. This made the app think it was standalone even though it wasn't.

Also, to make the web.config file write a .exe, we were passing in weird arguments (UseAppHost=true, Standalone=false) to force it to happen. By default, the web.config will never point to the portable exe. However, someone can manually add it. It really isn't 

This makes it so the test manually changes the web.config to point to the exe and removes the AppHost publish option as by default portable adds it now.

### CODE REVIEW

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs [hunk 1]
```diff
@@ -186,17 +185,27 @@ public async Task HelloWorld(TestVariant variant)
             await StartAsync(deploymentParameters);
         }
 
-        [SkipOnHelix] // https://github.com/aspnet/AspNetCore/issues/7972
         [ConditionalFact]
         [RequiresIIS(IISCapability.PoolEnvironmentVariables)]
         public async Task StartsWithPortableAndBootstraperExe()
         {
             var deploymentParameters = _fixture.GetBaseDeploymentParameters(_fixture.InProcessTestSite);
+            deploymentParameters.PublishApplicationBeforeDeployment = true;
```
Comment (pakrym) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs]: I don't think you need this.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs [hunk 2]
```diff
@@ -186,17 +185,27 @@ public async Task HelloWorld(TestVariant variant)
...
+
             // We need the right dotnet on the path in IIS
-            deploymentParameters.EnvironmentVariables["PATH"] = Path.GetDirectoryName(DotNetCommands.GetDotNetExecutable(deploymentParameters.RuntimeArchitecture));
             // ReferenceTestTasks is workaround for https://github.com/dotnet/sdk/issues/2482
-            deploymentParameters.AdditionalPublishParameters = "AppHost";
             var deploymentResult = await DeployAsync(deploymentParameters);
+            deploymentResult.ModifyWebConfig(element => element
```
Comment (pakrym) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs]: Use TransformPath/TransformArguments

File: src/Servers/IIS/IIS/test/testassets/InProcessWebSite/InProcessWebSite.csproj [hunk 3]
```diff
@@ -9,10 +9,9 @@
 
   <ItemGroup Condition="'$(OS)' == 'Windows_NT'">
     <!-- UseAppHost is a workaround for https://github.com/aspnet/websdk/issues/422 -->
```
Comment (pakrym) [inline src/Servers/IIS/IIS/test/testassets/InProcessWebSite/InProcessWebSite.csproj]: Remove.

**Comment by jkotalik** [comment]
wee woo wee woo @aspnet/brt https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190305.39/workItem/Microsoft.AspNetCore.Authentication.Test-netcoreapp3.0/wilogs for an authentication test. Other test failure was expected.

**Comment by jkotalik** [comment]
@pakrym do you remember seeing the following failure when enabling this test on helix?
```
[xUnit.net 00:09:09.80]       Expected: True
[xUnit.net 00:09:09.80]       Actual:   False) (Unable to find non-empty debug log files. Tried: C:\Users\runner\AppData\Local\Temp\tmp5EF.tmp, aspnetcore-debug.log)
[xUnit.net 00:09:09.80]       ---- Assert.True() Failure

```

**Comment by anurse** [comment]
Bee do bee do bee do

Not sure I understand, the link you provided says everything passed. It looks like only the "expected" failure is there...

**Comment by jkotalik** [comment]
Shoot, I didn't realize that when you rerun helix, it deletes the previous build.

File: src/Hosting/Server.IntegrationTesting/src/Common/DotNetCommands.cs [hunk 4]
```diff
@@ -16,15 +16,20 @@ public static class DotNetCommands
...
-            var dotnetHome = Environment.GetEnvironmentVariable("DOTNET_ROOT");
+            var dotnetHome = Environment.GetEnvironmentVariable("DOTNET_HOME");
+            var dotnetRoot = Environment.GetEnvironmentVariable("DOTNET_ROOT");
             var userProfile = Environment.GetEnvironmentVariable("USERPROFILE");
             var home = Environment.GetEnvironmentVariable("HOME");
 
             var result = Path.Combine(Directory.GetCurrentDirectory(), _dotnetFolderName);
             if (!string.IsNullOrEmpty(dotnetHome))
             {
                 // DOTNET_ROOT has x64 appended to the path, which we append again in GetDotNetInstallDir
```
Comment (pakrym) [inline src/Hosting/Server.IntegrationTesting/src/Common/DotNetCommands.cs]: Move the comment.

Comment (jkotalik) [reply src/Hosting/Server.IntegrationTesting/src/Common/DotNetCommands.cs]: Already did 😄

**Review by pakrym — APPROVED** [review]

**Review by pakrym — APPROVED** [review]

**Comment by jkotalik** [comment]
@aspnet/brt another inmemory process kill https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190306.18/workItem/InMemory.FunctionalTests-netcoreapp3.0/wilogs.

**Comment by anurse** [comment]
Short of skipping all of InMemory.FunctionalTests I'm not sure how to mitigate that. It may be one of the few categories that really are a truly high-pri fix...

**Pull request merged**